chain/chain/src/store.rs

```diff
@@ -2465,6 +2467,104 @@ impl<'a> ChainStoreUpdate<'a> {
         Ok(())
     }
 
+    /// Only used in mock network
+    /// Copy the necessary chain state related to `block_hash` from `source_store` to the current
+    /// store.
+    #[cfg(feature = "mock_network")]
+    pub fn copy_chain_state_as_of_block(
```

Comment:
Could we change it to constructor instead?  (so that it returns ChainStoreUpdate?)

(as using it on already partially created ChainStoreUpdate object might not be a good idea)